mautrix_telegram/__main__.py

```diff
@@ -117,8 +119,10 @@ async def start(self) -> None:
         if self.bot:
             try:
                 await self.bot.start()
-            except Exception as e:
+                METRIC_BOT_STARTUP_OK.set(1)
+            except telethon.errors.RPCError as e:
```

Comment:
This is the base class for telethon errors, I don't think we can get more specific than this given all the ways we've seen this failing.